likeminds-chat-reactnative-integration/ChatSX/customModals/ChatroomActionModal/index.tsx

```diff
@@ -0,0 +1,101 @@
+import { View, Text, Modal, Pressable, TouchableOpacity } from "react-native";
+import React from "react";
+import { styles } from "../../screens/ChatRoom/styles";
+import { ChatroomActions } from "../../enums";
+import {
+  ChatroomContextValues,
+  useChatroomContext,
+} from "../../context/ChatroomContext";
+import { onShare } from "../../shareUtils";
+
+const ChatroomActionModal = () => {
+  const {
+    chatroomID,
+    chatroomType,
+    chatroomDBDetails,
+    navigation,
+    isSecret,
+    filteredChatroomActions,
+    modalVisible,
+    setModalVisible,
+    handleModalClose,
+    showWarningModal,
+    joinChatroom,
+    muteNotifications,
+    unmuteNotifications,
+    handleBlockMember,
+    unblockMember,
+  }: ChatroomContextValues = useChatroomContext();
+  return (
+    <Modal
+      transparent={true}
+      visible={modalVisible}
+      onRequestClose={() => {
+        setModalVisible(!modalVisible);
+      }}
+    >
+      <Pressable style={styles.centeredView} onPress={handleModalClose}>
+        <View>
+          <Pressable onPress={() => {}} style={[styles.modalView]}>
+            {filteredChatroomActions?.map((val: any, index: any) => {
+              return (
+                <TouchableOpacity
+                  onPress={async () => {
+                    if (val?.id === ChatroomActions.VIEW_PARTICIPANTS) {
+                      setModalVisible(false);
+                      // navigation.navigate(VIEW_PARTICIPANTS, {
+                      //   chatroomID: chatroomID,
+                      //   isSecret: isSecret,
+                      //   chatroomName: chatroomName,
+                      // });
+                    } else if (
+                      val?.id === ChatroomActions.LEAVE_CHATROOM ||
+                      val?.id === ChatroomActions.LEAVE_SECRET_CHATROOM
+                    ) {
+                      // showWarningModal();
+                      setModalVisible(false);
+                    } else if (val?.id === ChatroomActions.JOIN_CHATROOM) {
+                      if (!isSecret) {
+                        // joinChatroom();
+                      }
+                      setModalVisible(false);
```

Comment:
can we put some sort of customizations, to show and hide three dot menu options?

Reply:
will cover it in callBacks customisations PR